include/xtensor/xbuffer_adaptor.hpp

```diff
@@ -378,7 +378,6 @@ namespace xt
         inline auto xbuffer_owner_storage<CP, A>::operator=(self_type&& rhs) -> self_type&
         {
             swap(rhs);
-            rhs.m_moved_from = true;
```

Comment:
this was a memory leak because the swapped to rhs didn't deallocate.

Reply:
👍 makes sense